backend/app/views/spree/admin/shared/_address_form.html.erb

```diff
@@ -6,6 +6,16 @@
     <%= f.text_field :name, class: 'fullwidth' %>
   </div>
 
+  <div class="field <%= "#{type}-row" %>">
```

Comment:
we cannot just introduce the two attributes as additional fields in the forms. How inconsistent is that? We need a toggle (or some very clever logic) to either show the two separate fields or the name field. But never both. Still I think just re-introducing the firstname as an option (via a configuration) is our best bet and helps migrating data. Please elaborate why you just introduced the two additional fields. Maybe I am missing something?